src/GovUk.Education.ExploreEducationStatistics.Data.Api/Services/PermalinkService.cs

```diff
@@ -258,15 +267,45 @@ private async Task WriteCsvHeaderRow(CsvWriter csv, PermalinkCsvMetaViewModel me
                         permalinkTableResult,
                         request.Query);
 
-                    await _blobStorageService.UploadAsJson(containerName: Permalinks,
+                    var content = JsonConvert.SerializeObject(legacyPermalink,
+                        BuildJsonSerializerSettings());
+
+                    await using var stream = new MemoryStream(Encoding.UTF8.GetBytes(content));
+                    permalink.LegacyContentLength = stream.Length;
```

Comment:
Interesting approach. Have we done this to avoid making a second request for the blob metadata?

Reply:
> Have we done this to avoid making a second request for the blob metadata?

Yes, whether we call `UploadStream`, or `UploadAsJson` which is turned into text and then to a stream, the underlying call used is `blob.UploadAsync` which returns `Response<BlobContentInfo>` but doesn't contain the length of the uploaded content.

> We seem to be okay getting the blob metadata as part of the migration, so just wondering why the inconsistent approach?

I thought it was ok to do it in the migration because we already get the blob and then just one extra call to get the blob properties and with it only being the migration it's code that's only going to run once under supervision and then be thrown away.

Here when creating a Permalink we upload the blob to begin with, then I think we'd need to get the blob again and then a further call to get the blob properties. It seems like that's adding unnecessary steps that could fail and result in us having a blob that's uploaded but not saved in the database.

I can work on trying to make this a bit nicer. I'm open to suggestions. Were you thinking I should change the `BlobStorageService` to try and return some kind of result object with all uploads? 🙂

Reply:
Fair enough, it doesn't seem like a problem to keep it the way it is. 

Personally, I think I would have just gone with two requests - one to upload it first, then another to fetch the properties afterwards. It doesn't really matter though as long as the content length from the stream accurately reflects what gets uploaded to blob storage!

Reply:
Thanks for reviewing. I did a check during some testing and the content length was identical to the content length reported in my local storage.